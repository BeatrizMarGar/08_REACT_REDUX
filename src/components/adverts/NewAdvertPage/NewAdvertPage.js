import React from 'react';
import T from 'prop-types';
import Layout from '../../layout';
import NewAdvertForm from './NewAdvertForm';
import { createAd } from '../../../store/actions';
import useStoreAction from '../../../hooks/StoreActions';

function NewAdvertPage() {
  const createnewad = useStoreAction(createAd)

  return (
    <Layout>
      <NewAdvertForm onSubmit={createnewad} />
    </Layout>
  );
}

export default NewAdvertPage;
