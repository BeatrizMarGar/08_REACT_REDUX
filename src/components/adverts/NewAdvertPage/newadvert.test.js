
import NewAdvertPage from './NewAdvertPage';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('NewAdvertPage', () => {
  test('snapshot test', () => {
    const store = {
      getState: () => ({ auth: true }),
      subscribe: () => {},
      dispatch: () => {},
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <NewAdvertPage />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
