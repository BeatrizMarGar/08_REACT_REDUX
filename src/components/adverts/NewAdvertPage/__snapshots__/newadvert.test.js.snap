// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewAdvertPage snapshot test 1`] = `
Array [
  <header>
    Nodepop React
    <nav>
      <ul>
        <li>
          <a
            aria-current={null}
            href="/adverts"
            onClick={[Function]}
          >
            Nodepop
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/adverts/new"
            onClick={[Function]}
          >
            New advert
          </a>
        </li>
      </ul>
    </nav>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </header>,
  <main>
    <form
      onSubmit={[Function]}
    >
      <input
        name="name"
        onChange={[Function]}
        value=""
      />
      <input
        checked={true}
        name="sale"
        onChange={[Function]}
        type="checkbox"
      />
      <input
        name="price"
        onChange={[Function]}
        type="number"
        value={0}
      />
      <div />
      <input
        name="photo"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
      <img
        alt=""
        height="200"
        onClick={[Function]}
        src="placeholder.png"
        style={
          Object {
            "objectFit": "contain",
          }
        }
        width="200"
      />
      <button
        disabled={true}
      >
        Save
      </button>
    </form>
  </main>,
  <footer>
    @Keepcoding 2021
  </footer>,
]
`;
